mon-pix/app/templates/components/navbar-header.hbs

```diff
@@ -1,7 +1,7 @@
 <div class="navbar-header">
   {{#if (media 'isDesktop')}}
-    <NavbarDesktopHeader />
+    <NavbarDesktopHeader @isFrenchDomain={{this.isFrenchDomainExtension}}/>
   {{else}}
-    <NavbarMobileHeader @burger={{@burger}} />
+    <NavbarMobileHeader @burger={{@burger}} @isFrenchDomain={{this.isFrenchDomainExtension}} />
```

Comment:
Ici, je trouve que l'on créé un couplage fort avec le composant parent.
Je pense qu'il faut soit :
- soit lui déléguer la responsabilité d'afficher la marianne mais ça impose un ré-usinage des navbar
- soit laisser l'entière responsabilité aux enfants d'aller chercher l'information d'environnement qui indique s'il faut ou non afficher la marianne.

De façon générale, passer un booléen en paramètre est un anti-pattern de clean code

Reply:
J'avais pas mal hésité ici, mais si l'on enlève l'argument `@isFrenchDomain={{this.isFrenchDomainExtension}}`, alors chaque enfant devra faire lui même l'appel, ce qui duplique le code ... =/

Reply:
Je suis moins gêné par la duplication de 3 lignes de codes que par un flag en paramètre. A voir avec les autres

Reply:
en renommant en `shouldShowMarianne` je pense que c'est ok.

Reply:
oui c'est renommé. Mais que penses-tu de la proposition de Stéphane d'éviter de mettre en argument `shouldShowMarianne` pour que chaque composant `navbarDesktopHeader` et `navbarDesktopHeader` le recupère lui-même ?

Reply:
ben du coup je suis contre, et je suis pour qu'on passe l'argument. Le composant est agnostique de la logique de locale & co.